### DISCUSSION TITLE
latest/detection/utils/

### DISCUSSION THREAD

**Discussion by giscus** [roboflow/supervision#1480] [Docs]
# latest/detection/utils/

A set of easy-to-use utilities that will come in handy in any computer vision project.

https://supervision.roboflow.com/latest/detection/utils/

**Comment by dearMOMO** [comment]
I want to determine whether the coordinate points are in an area, and I would like to ask if there are some cases for reference

**Reply by LinasKo** [reply]
Hi @dearMOMO,

We don't currently provide any utilities for that. I'll add it to the idea list!

**Comment by mendozajpd** [comment]
I want to be able to use the tools with my Webcam, is it possible? or is it only for videos only?

**Reply by LinasKo** [reply]
Hi @mendozajpd 👋 

Option 1 is to use InferencePiepline: https://inference.roboflow.com/quickstart/run_model_on_rtsp_webcam/

In supervision we don't have a wrapper around extracting images from a webcam feed, but you can very easily use normal OpenCV frames.

```
import cv2
from inference import get_model
import supervision as sv


model = get_model("yolov8n-640")
box_annotator = sv.BoxAnnotator()

cap = cv2.VideoCapture(0)

if not cap.isOpened():
    print("Cannot open camera")
    exit()

while True:
    ret, frame = cap.read()

    if not ret:
        print("Failed to grab frame")
        break

    result = model.infer(frame)[0]
    detections = sv.Detections.from_inference(result)
    annotated_frame = box_annotator.annotate(frame.copy(), detections)
    
    cv2.imshow('Webcam Stream', annotated_frame)
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

cap.release()
cv2.destroyAllWindows()
```

Try this - I believe it will work 😉

**Comment by Seredeep** [comment]
Is an API key needed to run inference locally? Why? What happens if I run in airgapped hardware?

**Reply by LinasKo** [reply]
The model needs to be downloaded at least once. My understanding is that we don't support 100% offline mode at the moment, even if you have weights trained on another system. It should work if you do it once and then go offline, but I haven't personally tested it.

**Reply by Seredeep** [reply]
so much for open source...

**Comment by amkbrys** [comment]
how to count objects on the image captured through the mobile app

**Reply by LinasKo** [reply]
Hi @amkbrys 👋 

I'm less familiar with that side of our product, but you should be able to parse the response from one of these calls: https://docs.roboflow.com/deploy/hosted-api/custom-models/object-detection

Here are the links if you need other tasks: https://docs.roboflow.com/api-reference/inference

**Comment by NicolasGonzalezF** [comment]
Questions about licensing, im interested in using supervision for object detection and training new datasets but im confused, in some instances for detections ultralytics is used but they have license tipe AGLP 3.0 and supervisions is MIT , how does that work in a comercial env ?

**Comment by AlgosDuCouteau** [comment]
can supervision run with yolov11?

**Reply by onuralpszr** [reply]
> can supervision run with yolov11?

Yes it can

**Reply by LinasKo** [reply]
To expand on that - absolutely!

Here's some example code to get you started:

```python
import cv2
import supervision as sv
from inference import get_model

image = cv2.imread("dog.jpeg")
model = get_model(model_id="yolo11n-640")
results = model.infer(image)[0]
detections = sv.Detections.from_inference(results)

annotated_image = sv.BoxAnnotator().annotate(
    scene=image.copy(), detections=detections
)
annotated_image = sv.LabelAnnotator().annotate(
    scene=annotated_image, detections=detections
)
sv.plot_image(annotated_image)
```

* Available pre-trained models: [Inference Docs - "Models: Popular"](https://inference.roboflow.com/quickstart/aliases/)
* Get the image by running `wget https://media.roboflow.com/notebooks/examples/dog.jpeg`

**Comment by gotfrid99** [comment]
Hi, i want to know whether  i can get the coordinates of the boxes using Supervision

**Reply by LinasKo** [reply]
Hi @gotfrid99 👋 

`detections.xyxy` will give you the coordinates of top-left and bottom-right corners, for every detected object. See [Detections](https://supervision.roboflow.com/latest/detection/core/).

**Comment by AlgosDuCouteau** [comment]
hi I want to implement an application that count in/out when an object passes through a line, eg. move from left side of the line to the right side of the line is in and reverse is out. Is there any reference for that?

**Reply by LinasKo** [reply]
Hi @AlgosDuCouteau 👋

That's exactly what LineZone does. Check out the example in the docs: https://supervision.roboflow.com/latest/detection/tools/line_zone/

You may check out this video as well: https://youtu.be/OS5qI9YBkfk?si=BGTVgxTMptDnPESv&t=1052

**Comment by b0bursoliyev** [comment]
Hi I'm fresher at computer vision and first time using supervision and I have that problem the line is not working, can anyone help me
https://drive.google.com/file/d/1Os19dW-MoxquC9bMwcNOL94RWG5uCTfq/view?usp=sharing 
https://drive.google.com/file/d/1VyF9anMSB4ksArVxFm7rQkUNOySmqFnc/view?usp=sharing

**Reply by LinasKo** [reply]
Hi @b0bursoliyev 👋 

You may find these two issues helpful:
* https://github.com/Deci-AI/super-gradients/issues/2064
* https://github.com/Deci-AI/super-gradients/issues/2057

YOLO-NAS weights were removed from their original locations.

**Comment by nicstark** [comment]
Hi @LinasKo,

Using Supervision and Yolo11 I'm running into an issue where cyclists are being labelled as two separate objects, a bicycle and a person. Is there a simple way to remedy this without having to retrain the model? 

Thanks,

Nic

**Reply by LinasKo** [reply]
Hi Nic,

Unfortunately, we don't have it out-the-box. Indeed, the foolproof solution is model retraining, yet I'd also try merging individual detections, having checked that bicycle is below a person.

**Comment by HugoSun** [comment]
I want the Inference pipeline to process and count objects crossing the line. How can I do this?

**Comment by HugoSun** [comment]
can supervision run with C++?

**Reply by LinasKo** [reply]
Nope. We're solely a Python library.
